lso/applications/objective_functions/squares/squares_area.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+from typing import Optional
+
+from lso.utils import area_utils
+from lso.data import data as lso_data
+from lso.objective_function import objective_function as lso_objective_function
+
+
+class SquaresArea(lso_objective_function.ObjectiveFunction):
+
+    def __init__(self, shape_likeliness_coeff: Optional[float] = 1.0):
+        self.shape_likeliness_coeff = shape_likeliness_coeff
+
+    def calculate_shape_likeliness(self,
+                                   x: np.array,
+                                   metric: Optional[str] = 'Jaccard') -> np.array:
+
+        if metric == 'Jaccard':
+
+            # Compute the Jaccard distance for areas of: shape x ,and the square circumscribed on shape x.
+
+            area_x = area_utils.calculate_area(x)
+            area_circumscribed_x = area_utils.calculate_circumscribed_square_area(x)
+
+            jaccard_index = area_x / area_circumscribed_x
+            shape_likeliness = 1 - jaccard_index
+
+        if metric not in ['Jaccard']:
+            raise ValueError('provide a correct metric.')
+
+        return shape_likeliness
+
+    def evaluate(self, data: lso_data.Data) -> lso_data.Data:
+
+        assert issubclass(type(data), lso_data.Data)
+        assert len(data.x.shape) == 3
```

Comment:
This is a low level assertion. It is might also be misleading and error prone in the future - as e.g., one might provide images in shape (BatchDim, W, H, C=1)

Reply:
It is necessary for the method to output correct results.

Reply:
What I could do is to create a new object RectangleData which assumes that len(x.shape) == 3, let me know if this is a good idea.